### PR TITLE
Add docs

### PR DESCRIPTION
**Pull request by mgoulao**
# Description

Add the documentation folder. I will add the workflow to deploy the docs website to gh-pages, and the versioning in a separate PR.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ repos:
     hooks:
       - id: codespell
         args:
-          - --ignore-words-list=nd,reacher,thist,ths, ure, referenc
+          - --ignore-words-list=nd,reacher,thist,ths, ure, referenc,Wile
```
Comment (pseudo-rnd-thoughts) [inline .pre-commit-config.yaml:12]: I believe this needs to be lower case

**Review by pseudo-rnd-thoughts — CHANGES_REQUESTED** [review]

**Comment by pseudo-rnd-thoughts** [comment]
Do we need to modify the build process to not include the docs?

File: .pre-commit-config.yaml (hunk 1 above)
Comment (mgoulao) [reply .pre-commit-config.yaml:12]: ok, that makes sense

**Comment by mgoulao** [comment]
> Do we need to modify the build process to not include the docs?

I don't think so, both `tests` and `docs` are ignored.

**Comment by pseudo-rnd-thoughts** [comment]
Do we want to add some pre-commit for markdown? https://github.com/markdownlint/markdownlint

**Comment by mgoulao** [comment]
> Do we want to add some pre-commit for markdown? https://github.com/markdownlint/markdownlint

We can, but I think it's better to leave that to a future PR.

**Review by pseudo-rnd-thoughts — APPROVED** [review]

**Review by pseudo-rnd-thoughts — APPROVED** [review]

**Pull request merged**